### PR TITLE
controlpanel: CloneNetwork functionality

### PR DESCRIPTION
**Pull request by kashike**

### CODE REVIEW

File: modules/controlpanel.cpp [hunk 1]
```diff
@@ -619,6 +619,35 @@ class CAdminMod : public CModule {
...
+        }
+
+        CIRCNetwork* pOldNetwork = FindNetwork(pUser, sOldNetwork);
+        if (!pOldNetwork) {
+            PutModule("Error: Network [" + sOldNetwork + "] not found.");
+            return;
+        }
+
+        CIRCNetwork* pNewNetwork = new CIRCNetwork(pUser, *pOldNetwork);
+        pNewNetwork->SetName(sNewNetwork);
```
Comment (DarthGandalf) [inline modules/controlpanel.cpp]: This won't work if new name contains invalid characters.

**Comment by DarthGandalf** [comment]
Need to check `pUser->HasSpaceForNewNetwork()` as AddNetwork does, otherwise this will be used to workaround limits set by administrator.

**Comment by kashike** [comment]
Is this what you meant?

File: modules/controlpanel.cpp (hunk 1 above)
Comment (psychon) [inline modules/controlpanel.cpp]: > Is this what you meant?

`SetName` can fail. Please look at the following and handle this function's return value:
https://github.com/znc/znc/blob/f885699d1a3cb6376dc6162f9e8440791034a329/src/IRCNetwork.cpp#L771-L778

Comment (kashike) [inline modules/controlpanel.cpp]: https://github.com/znc/znc/pull/1434/files#diff-29017c32fc7e54eb3ad373b49ef002f1R633

**Comment by DarthGandalf** [comment]
What happens if network with the new name exists already?
Any test for this?

**Pull request closed without merging**